### PR TITLE
enhance the module system + allow devs to customize interface/better DOM architecture

### PR DESCRIPTION
**Pull request by YglesEyes**

### CODE REVIEW

File: core/client/ui/radial-menu.js [hunk 1]
```diff
@@ -131,7 +131,13 @@ const buildMenuFromOptions = options => {
 const onMouseMove = event => {
   if (!Session.get('menu') || menuOpenUsingKey) return;
   const menuPosition = computeMenuPosition();
-  const mousePosition = { x: event.clientX, y: event.clientY };
+
+  // Get canvas bound go manage mouse position offset
+  const canvasBounds = document.querySelector('#game canvas').getBoundingClientRect();
```
Comment (Donorhan) [inline core/client/ui/radial-menu.js:136]: Is it not possible to cache this? A call to the DOM at each mouse movement is very expensive

File: core/client/helpers.js [hunk 2]
```diff
@@ -207,6 +208,8 @@ const addToSession = (key, values) => {
 };
 
 registerModules = modules => addToSession('modules', modules);
+registerMainModules = modules => addToSession('mainModules', modules);
+registerGameModules = modules => addToSession('gameModules', modules);
```
Comment (Donorhan) [inline core/client/helpers.js:212]: It's getting complicated I think, I don't see the difference between "modules" and "mainModules" 🤔. If it was just to have a different interface wouldn't it be better to just add a class on the parent selector?

File: core/client/lemverse.js [hunk 3]
```diff
@@ -469,6 +469,9 @@ Template.lemverse.helpers({
   screenMode: () => Template.instance().screenMode.get(),
   settingsOpen: () => (!Session.get('modal') ? false : (Session.get('modal').template.indexOf('settings') !== -1)),
   modules: () => Session.get('modules'),
+  mainModules: () => Session.get('mainModules'),
+  gameModules: () => Session.get('gameModules'),
+  displayNotificationButton: () => (Meteor?.settings?.public?.features?.notificationButton?.enabled !== false),
```
Comment (Donorhan) [inline core/client/lemverse.js:474]: I think you copied this code somewhere else but it doesn't seem useful to me all the "?", in any case Meteor.settings.public will necessarily exist.

File: core/client/lemverse.scss [hunk 4]
```diff
@@ -274,3 +274,28 @@ button.js-notifications {
     background-color: darken($light-orange, 10%);
   }
 }
+
+/* width */
+::-webkit-scrollbar {
```
Comment (Donorhan) [inline core/client/lemverse.scss:280]: Nice 🙏🏻

**Review by Donorhan — CHANGES_REQUESTED** [review]

File: core/client/helpers.js (hunk 2 above)
Comment (YglesEyes) [reply core/client/helpers.js:212]: Component are not inserted at the same place into the DOM

**Review by Donorhan — APPROVED** [review]

**Pull request merged**